sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/Regex.java

```diff
@@ -71,7 +138,46 @@ public static Matches matches(String regex, int group) {
    * @param valueGroup The Regex group to use the value
    */
   public static MatchesKV matchesKV(String regex, int keyGroup, int valueGroup) {
-    return new MatchesKV(regex, keyGroup, valueGroup);
+    return matchesKV(Pattern.compile(regex), keyGroup, valueGroup);
+  }
+
+  /**
+   * Returns a {@link Regex.MatchesKV} {@link PTransform} that checks if the entire line matches the
+   * Regex. Returns the specified groups as the key and value as a {@link PCollection}.
+   *
+   * @param pattern The regular expression to run
+   * @param keyGroup The Regex group to use as the key
+   * @param valueGroup The Regex group to use the value
+   */
+  public static MatchesKV matchesKV(Pattern pattern, int keyGroup, int valueGroup) {
+    return new MatchesKV(pattern, keyGroup, valueGroup);
+  }
+
+  /**
+   * Returns a {@link Regex.MatchesNameKV} {@link PTransform} that checks if the
+   * entire line matches the Regex. Returns the specified groups as the key and value as a
+   * {@link PCollection}.
+   *
+   * @param regex The regular expression to run
+   * @param keyGroupName The Regex group name to use as the key
+   * @param valueGroupName The Regex group name to use the value
+   */
+  public static MatchesNameKV matchesKV(String regex, String keyGroupName, String valueGroupName) {
```

Comment:
Yes, they often have a regex that they've been using. Usually there's some kind of ETL regex that they're reusing. The `int` based version is used when the order for the KV is inverted in the regex. e.g. the value is group 1 and the key is group 2.